naima/models.py

```diff
@@ -431,3 +431,60 @@ def __call__(self, e):
         e = _validate_ene(e)
         interpy = np.power(10, self._interplogy(np.log10(e.to('eV').value)))
         return self.amplitude * interpy * self.unit
+      
+      
+class EblAbsorptionModel(TableModel):
+    """
+    A TableModel containing the different absorption values from a specific model.
+
+    It returns dimensionless opacity values, that could be multiplied to any model.
+    
+    Parameters
+    ----------    
+    redshift : float
+        Redshift considered for the absorption evaluation.
+        The minimum value used is z = 0.01; Lower values will
+        be considered as redshift 0.
+        
+    ebl_absorption_model : string 
+        Name of the EBL absorption model to use. It may be:
+    
+    * A string equal to ``Dominguez`` (default), containing Dominguez 2011 EBL model
+    
+    NOTE: Current implementation does NOT perform an interpolation in the redshift, so 
+    it just uses the closes z value from the finely binned tau_dominguez11.out file 
+    (delta_z=0.01)    
+        
+    """
+
+    def __init__(self, redshift, ebl_absorption_model='Dominguez'):
+        from scipy.interpolate import interp1d
+        import os
+  
+        self.redshift = validate_scalar('redshift',
+                                   redshift.value,
+                                   domain='positive')
+        # check that model has allowed value
+        if ebl_absorption_model not in set(['Dominguez',]):
+          raise ArgumentError("Model should be one of [Dominguez, ]")
+        self.model = ebl_absorption_model
+        
+        if self.model == 'Dominguez':
+            taus_table = np.load(os.path.join(os.path.dirname(__file__), 'data/tau_dominguez11.npz'))['arr_0']
+            redshift_list = np.arange(0.01,4,0.01)
+            energy = taus_table['energy'] * u.TeV
+            if (self.redshift >= 0.01):
+                taus = 10**taus_table['col%s' % (2+(np.abs(redshift_list-self.redshift)).argmin())] * u.dimensionless_unscaled
+            elif (self.redshift < 0.01): 
+                taus = 10**np.zeros(len(taus_table['energy'])) * u.dimensionless_unscaled
+        super(EblAbsorptionModel, self).__init__(energy, taus)
+        
+      
+    def transmission(self, data):
+        taus = np.zeros(len(data['energy']))
+        for i in range(0,len(data['energy'])):
+            if data['energy'][i].to('GeV').value < 1.: taus[i] = 0.
+            elif data['energy'][i].to('TeV').value > 100. : taus[i] = np.log10(6000.)
+            else: taus[i] = np.log10(super(EblAbsorptionModel,self).__call__(data['energy'][i]))
```

Comment:
You don't need to use `super` here, you can just do `self.__call__`. It is needed in the init because we are overriding the init in the subclass, but since we are not overrriding `__call__`, using `self.__call__` will already point to the superclass definition.